### PR TITLE
v1.6.0

### PR DESCRIPTION
**Pull request by grahamhency**
###### Change log for `v1.6.0`

***

# ⚠️ Breaking Changes

### Complete Refactor of `GoConfigService`

- GoConfigService
   - Removed the toggleHeaderBrandingEnabled method in favor of setConfig
   - Removed the overrideMenuColor method
   - Moved the contrastIsAccessible method to the new colors.util.ts file
   - Moved the RGB interface to the new colors.util.ts file
- GoHeader Component
   - Removed bindings for altText, logo, logoLink, and logoLinkUseRouterLink in favor of setting them in the setConfig method on the GoConfigService
- GoConfigInterface
   - Removed the brandFontColor property as it is no longer necessary
   - Removed the headerBrandingEnabled property as it has been replaced by the brandingMode property

###### The reasons for this are:
This approach is a much simpler implementation for branding and other configurations within applications. Most of these will not effect implementations.

###### Changes Required
1. Change anywhere that `toggleHeaderBrandingEnabled` method was used to instead use the new `setConfig` method
2. Where `<go-header>` is used, remove any of these bindings (`altText`, `logo`, `logoLink`, and `logoLinkUseRouterLink`). Then when the initial configuration is set, these bindings can be set through using the `setConfig` method. (This will be done in the component where the `go-layout` component resides)
3. Remove any reference to `brandFontColor`
4. Change `headerBrandingEnabled` property to `brandingMode`. This may not need to be done if the new "client" branding mode is not implemented
5. Remove any reference to `overrideMenuColor`

### It is strongly recommended to [read the new documentation]() around this service for full understanding of how it works now. Also, for further reference or troubleshooting [the PR](https://github.com/mobi/goponents/pull/596) could be helpful.

***

# 🚢 New Features

### Action Sheet
- Added the ability to do content projection inside of a `go-panel` - #619 

### Config Service
- Added multiple features and methods - #596 
   - Added ability to specify a secondary logo 
   - Added the ability to `setConfig`
   - Added the ability to `getConfig`
   - Added the ability to `setLogo`
   - Added the ability to `setBrandingMode`

### Copy Card Link
- Added a new directive to copy a link to the card it resides in - #607 

### File Upload
- Added the ability to reset the component - #610 

### Select
- Added the ability to specify a template for `optgroup` - #621 
- Added the ability to specify text for typeahead placeholder - #624 

### Side Nav
- Added the ability to pass custom `target`s for external links - #622 

### Table
- Added the ability to specify a template for the title area - #599 

### Tree
- Added a new component for formatting a tree-like structure - #420

### Utilities
- Added a new `colors.util.ts` file - #596 
   - `RGB` interface
   - `contrastIsAccessible` method
   - `shadeHex` method
   - `shadeRGB` method
   - `rgbToHex` method
   - `hexToRgb` method
   - `luminance` method
   - `calculateContrast` method

### Virtual Scroll
- Added documentation for implementing virtual scroll (Material CDK) - #629 

# 🐞 Bug Fixes

### Form Elements
- Fixed an issue with various form elements with incorrect height alignment - #598 

### File Upload
- Fixed an issue that did not allow for re-uploading the same file - #610 

### Select
- Fixed an issue where the `title` attribute on the label template would not appear - #623 

# ✏️ General & Design

### Angular 8
This version of Goponents introduces support for Angular version 8, this also means we're upgrading the `ng-select` peer dependency for the `go-select`. Related PR's: #590, #597 

### Style Guide
Added links to every card example in the documentation, for easy reference - #601 

### Misc
- Added user feedback to `go-copy` - #620 
- Corrected color variables for light bg - #626 
- Incremented lodash version for security vulnerability - #627 
- Updated `go-action-sheet` styles - #625 
- Updated `go-datepicker` styles - #609 
- Updated `go-select` options styles - #606 
- Updated `go-side-nav` styles - #617 
- Updated Horizon and Nebula hex colors - #608

### CODE REVIEW

File: projects/go-lib/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "description": "A library of UI components for Angular v7+.",
   "license": "MIT",
   "peerDependencies": {
-    "@ng-select/ng-select": "^2.20.5"
+    "@ng-select/ng-select": "~3.7.3"
   },
   "devDependencies": {
     "@angular/common": "^7.0.0",
```
Comment (StevenUlmer) [inline projects/go-lib/package.json:15]: Do we need to update these dependencies to be angular 8?

Comment (eligolding) [reply projects/go-lib/package.json:15]: We should give a range, as I think we are compatible with multiple versions.

Comment (eligolding) [reply projects/go-lib/package.json:15]: oh woops, these are dev dependencies, not peer. Yeah it should be 8.

Comment (grahamhency) [reply projects/go-lib/package.json:15]: @eligolding #632

File: projects/go-lib/src/lib/components/go-copy/go-copy.component.spec.ts [hunk 2]
```diff
@@ -49,6 +52,19 @@ describe('goCopyComponent', () => {
 
       expect(document.execCommand).toHaveBeenCalledWith('copy');
     });
+
+    it('sets the icon to a check mark for three seconds', fakeAsync(() => {
```
[the remark this answers is not in the document]
Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-copy/go-copy.component.spec.ts]: Thanks, I fixed it here #633

**Comment by grahamhency** [comment]
For some reason Travis is not reporting builds right now. I can confirm [the build](https://travis-ci.org/github/mobi/goponents/builds/711503139) is successful and we're good to merge this.

**Pull request merged**